ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-interactions/index.tsx

```diff
@@ -298,114 +225,194 @@ const createResultActionsExportRegion = (model: Model) =>
+      icon="fa fa-download"
+      onClick={() => handleDownload(props.model)}
+    >
+      {isRemoteResourceCached(props.model) && (
+        <span
+          data-help="Displayed if the remote resource has been cached locally."
+          className="download-cached"
+        >
+          Local
+        </span>
+      )}
+    </MetacardInteraction>
+  )
+}
+
+const hasLocation = (model: Model): boolean => getGeoLocations(model).length > 0
+
+const CreateLocationSearch = (props: any) => {
+  if (!hasLocation(props.model)) {
+    return null
+  }
+  return (
+    <MetacardInteraction
+      {...props}
+      icon="fa fa-globe"
+      text="Create Search from Location"
+      help="Uses the geometry of the metacard to populate a search"
+      onClick={() => handleCreateSearch(props)}
+    />
+  )
+}
+
+const ExportActions = (props: any) => {
+  return (
+    <MarionetteRegionContainer
+      data-help="Opens the available actions for the item."
+      className="metacard-interaction interaction-actions-export composed-menu"
+      view={createResultActionsExportRegion(props.model)}
+      viewOptions={{ model: props.model }}
+    />
+  )
+}
 
-    const toggleIsBlacklisted = () =>
-      this.props.el.toggleClass(
-        'is-blacklisted',
-        isBlacklisted(this.props.model)
-      )
+const BlacklistToggle = (props: any) => {
+  if (props.blacklisted) {
+    return (
+      <MetacardInteraction
+        text="Unhide from Future Searches"
+        help="Removes from the
+              list of results that are hidden from future searches."
+        icon="fa fa-eye"
+        onClick={() => handleShow(props)}
+      />
+    )
+  } else {
+    return (
+      <MetacardInteraction
+        text="Hide from Future Searches"
+        help="Adds to a list
+              of results that will be hidden from future searches.  To clear this list,
+              click the Settings icon, select Hidden, then choose to unhide the record."
+        icon="fa fa-eye-slash"
+        onClick={() => handleHide(props)}
+      />
+    )
+  }
+}
+
+const ExpandMetacard = (props: any) => {
+  const isRouted = router && router.toJSON().name === 'openMetacard'
+
+  if (isRouted || props.model.length > 1) {
+    return null
+  }
+
+  return (
+    <MetacardInteraction
+      text="Expand Metacard View"
+      help="Takes you to a
+            view that only focuses on this particular result.  Bookmarking it will allow
+            you to come back to this result directly."
+      icon="fa fa-expand"
+      onClick={() => handleExpand(props)}
+    />
+  )
+}
+
+type State = {
+  blacklisted: Boolean
+  model: any
+}
+
+const interactions = plugin([
+  AddToList,
+  BlacklistToggle,
+  ExpandMetacard,
+  Divider,
+  DownloadProduct,
+  CreateLocationSearch,
+  ExportActions,
+])
+
+class MetacardInteractions extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
```

Comment:
When I'm dealing with translating Backbone over to State, I like going with this pattern:
```
this.state = mapPropsToState(props)
```
In this case that function would be:
```
const mapPropsToState = (props: Props) => {
    return {
        model: props.model,
        blacklisted: isBlacklisted(props.model)
    }
}
```

Then whenever I need to update the state I can call:
```
this.setState(mapPropsToState(this.props))
```

Sometimes it makes things simpler to deal with.

Reply:
Changed